added ESPino to supported boards

Rodmg opened the pull request:
Added support for the ESPino board (http://www.espino.io/en) in boards.txt and variants/espino.

igrr commented on boards.txt:
```diff
@@ -691,3 +691,88 @@ d1_mini.menu.FlashSize.4M1M.build.spiffs_start=0x300000
+espino.menu.FlashFreq.80=80MHz
+espino.menu.FlashFreq.80.build.flash_freq=80
+
+espino.menu.FlashMode.dio=DIO
```
Since you probably know the exact flash model used by this board, DIO/QIO and flash frequency selection is probably unnecessary?

igrr commented:
Could you please add this board to doc/boards.md and to package/package_esp8266com_index.template.json as well?

Rodmg commented:
@igrr, Thank you for your remarks, I added the requested changes. About the flash frequency and mode, I removed the frecuency selection as it is always 40, but I think keeping flash mode selection could be handy as future versions of this board will come with the ESP-12E module. I'm not completely sure but I've read somewhere that changing flash mode frees some of the extra pins of that module?
